src/blockrelay/graphene_set.cpp

```diff
@@ -31,6 +31,14 @@ CGrapheneSet::CGrapheneSet(size_t _nReceiverUniverseItems,
     // Below is the parameter "n" from the graphene paper
     uint64_t nItems = _itemHashes.size();
 
+    uint64_t ibltVersion;
```

Comment:
There is also code duplication with the graphene -> IBLT version mapping in `graphene_set.h`.

Reply:
For this and other comments on version, I create static methods in CGrapheneSet and CGrapheneBlock that, given the version of those objects, would return the correct version for the corresponding CIblt and CGrapheneSet objects, respectively. This seems to be a lot cleaner and, even though the version logic is confusing, at least it's all in one place now. Does that work for you?